src/Solvers/ihs.jl

```diff
@@ -0,0 +1,234 @@
+"""
+    IHS <: Solver
+
+An implementation of the Iterative Hessian Sketch solver for solving over determined 
+least squares problems (@cite)[pilanci2014iterative].
+ 
+# Mathematical Description
+Let ``A  \\in \\mathbb{R}^{m \\times n}`` and consider the least square problem ``\\min_x 
+\\|Ax - b \\|_2^2``. If we let ``S \\in \\mathbb{R}^{s \\times m}`` be a compression matrix, then 
+Iterative Hessian Sketch iteratively finds a solution to this problem
+by repeatedly updating ``x_{k+1} = x_k + \\alpha u_k``where ``u_k`` is the solution to the 
+convex optimization problem, 
+``u_k = \\min_u \\{\\|S_k Au\\|_2^2 - \\langle A, b - Ax_k \\rangle \\}.`` This method 
+has been to shown to converge geometrically at a rate ``\\rho \\in (0, 1/2]``, typically the 
+required compression dimension needs to be 4-8 times the size of n for the algorithm to 
+perform successfully.
+
+# Fields
+- `alpha::Float64`, a step size parameter.
+- `compressor::Compressor`, a technique for forming the compressed linear system.
+- `log::Logger`, a technique for logging the progress of the solver.
+- `error::SolverError`, a method for estimating the progress of the solver.
+
+# Constructor
+    function IHS(;
+        compressor::Compressor = SparseSign(cardinality = Left()),
+        log::Logger = BasicLogger(),
+        error::SolverError = FullResidual(),
+        alpha::Float64 = 1.0
+    )
+## Keywords
+- `compressor::Compressor`, a technique for forming the compressed linear system.
+- `log::Logger`, a technique for logging the progress of the solver.
+- `error::SolverError', a method for estimating the progress of the solver.
+- `alpha::Float64`, a step size parameter.
+
```

Comment:
Throw in a warning or an information admonition with some guidance about the value of `alpha`